File: Codeforces/cf1418A.cpp
#include <algorithm>
#include <cctype>
#include <cmath>
#include <cstdio>
#include <cstring>
#include <iostream>
#include <map>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <unordered_map>
#include <vector>
#define mst(a, b) memset((a), (b), sizeof(a))
#define debug(x) cerr << #x << " = " << x << "\n"
#define INF 0x3f3f3f3f
#define lson lef, mid, rt << 1
#define rson mid + 1, rig, rt << 1 | 1
const int maxn = 1e5 + 5;
using namespace std;
typedef long long ll;
typedef unsigned long long ull;
ll Ceil(ll a, ll b)
{
    return (a + b - 1) / b;
}
int main()
{
    int T;
    scanf("%d", &T);
    while (T--) {
        ll x, y, k;
        cin >> x >> y >> k;
        ll up = k + y * k - 1LL;
        ll down = x - 1LL;
        ll ans = Ceil(up, down) + k;
        printf("%lld\n", ans);
    }
    return 0;
}

File: Codeforces/cf1418A.py
import math

T = int(input())
while T > 0:
    T = T - 1
    x, y, k = [int(x) for x in input().split(' ')]
    up = k + y * k - 1
    down = x - 1
    tmp = up / down
    ans = math.ceil(tmp) + k
    if tmp * down < up:
        ans = ans + 1
    elif tmp * down > up:
        ans = ans - 1
    print(int(ans))
